articles/azure-app-configuration/enable-dynamic-configuration-dotnet-core-push-refresh.md

```diff
@@ -175,6 +175,74 @@ A random delay is added before the cached value is marked as dirty to reduce pot
 > [!NOTE]
 > To reduce the number of requests to App Configuration when using push refresh, it is important to call `SetCacheExpiration(TimeSpan cacheExpiration)` with an appropriate value of `cacheExpiration` parameter. This controls the cache expiration time for pull refresh and can be used as a safety net in case there is an issue with the Event subscription or the Service Bus subscription. The recommended value is `TimeSpan.FromDays(30)`.
 
+## Optional: Incorporate Synchronization tokens for real-time consistency
+
+> [!NOTE]
+> The API's used in this example are only available in the provider library version 5.x and later.
+
+Due to the nature of distributed systems, real-time consistency between requests is difficult to enforce implicitly. We use synchronization tokens to address this issue. The Event Grid event contains the sync-token that can be utilized to guarantee real-time consistency between different client instances and requests.
+
+The implementation of this PushModel introduces one new `PushNotification` class and two new API's:
+
+**PushNotification**
+```cs
+public class pushNotification() {
+    public Uri ResourceUri;
+    public string SyncToken;
+    public string EventType;
+}
+```
+This class stores the `ResourceUri` of the resource which triggered the `PushNotification`, the `Synchronization Token` to be added to the next request to the App Configuration Service, and the Type of Event which triggered the `PushNotification`.
+
+**TryCreatePushNotification**
+```cs
+EventGridEvent.TryCreatePushNotification(out PushNotification pushNotification)
+```
+This method uses the `Data`, `EventType`, and `Subject` stored in the Event Grid Event to try and create a `PushNotification`. If successful returns *true* and an out `PushNotification` parameter. Otherwise the parameter will be null and the method will return *false*.
+
+**ProcessPushNotification**
+```cs
+ProcessPushNotification(PushNotification pushNotification, TimeSpan? maxDelay)
+```
+This method is called by the refresher and provider to process the `PushNotification`. `ProcessPushNotification()` will invoke `UpdateSyncToken()` and then invoke `SetDirty()` to ensure the most recent configuration is provided to the user.
+
+**Implementation**
```

Comment:
Everything from 185-209 can be removed. We dont need to elaborate on the APIs in this document.